sedaro/src/sedaro/grpc_client/cosim_client.py

```diff
@@ -0,0 +1,257 @@
+import asyncio
+import base64
+import itertools
+import json
+import logging
+import os
+import uuid
+from typing import Any, Coroutine, Optional, Tuple
+
+import aiohttp
+import grpc
+
+from ..utils import serdes
+from . import cosim_pb2, cosim_pb2_grpc
+
+REFRESH_INTERVAL = 60 * 4
+
+
+class MetadataClientInterceptor(grpc.aio.UnaryUnaryClientInterceptor):
+    def __init__(self):
+        self.metadata_to_add = None
+
+    def set_uuid(self, uuid: str):
+        print(f"Setting UUID: {uuid}")
+        self.metadata_to_add = [("session-uuid", uuid)]
+
+    def _add_metadata(self, client_call_details):
+        metadata = list(client_call_details.metadata) if client_call_details.metadata else []
+        if self.metadata_to_add:
+            metadata.extend(self.metadata_to_add)
+        return client_call_details._replace(metadata=metadata)
+
+    async def intercept_unary_unary(self, continuation, client_call_details, request):
+        new_details = self._add_metadata(client_call_details)
+        print(f"Adding metadata: {new_details.metadata}")
```

Comment:
Should the be removed or was it intentionally left in?  Same question re: line 24 above.

Reply:
I think this might be outdated so you can ignore/resolve